src/util/errorHandling/errorHandler.ts

```diff
@@ -0,0 +1,35 @@
+import { Request, Response } from "express";
+import { ApiError } from "./apiError";
+import { HttpResponseCode } from "./httpResponseCode";
+import { NextFunction } from "connect";
+
+const toEmails: string[] = ["admin@unicsmcr.com"];
+
+/**
+ * Handles errors thrown by requests
+ */
+export const errorHandler = (err: ApiError|Error, req: Request, res: Response, next: NextFunction) => {
+  if (err instanceof Error) {
+    if (process.env.ENVIRONMENT === "production") {
+      // Send notification to admins when an uncaught error occurs
+    //   sendEmail("noreply@unicsmcr.com",
+    //   toEmails,
+    //     "Uncaught Error: " + err.name,
+    //     err.message + err.stack
+    //   );
+    }
+
+    console.error(err.stack);
+    res.status(HttpResponseCode.INTERNAL_ERROR).send(new ApiError(HttpResponseCode.INTERNAL_ERROR, err.stack));
```

Comment:
Might not be a great idea to send the entire error stack trace to the user in prod as it might contain some sensitive information.

Reply:
Good catch, I'll change this